build/Build.cs

```diff
@@ -0,0 +1,1045 @@
+        .DependsOn(PackDebianPackage)
+        .DependsOn(PackRedHatPackage)
+        .Executes(() =>
+        {
+            void RunLinuxPackageTestsFor(IReadOnlyList<string> testConfiguration)
+            {
+                var framework = testConfiguration[0];
+                var runtimeId = testConfiguration[1];
+                var dockerImage = testConfiguration[2];
+                var packageType = testConfiguration[3];
+
+                InTest($"{framework}/{runtimeId}/{dockerImage}/{packageType}", () =>
+                {
+                    string? archSuffix = null;
+                    if (packageType == "deb")
+                    {
+                        if (runtimeId == "linux-x64") archSuffix = "_amd64";
+                    }
+                    else if (packageType == "rpm")
+                    {
+                        if (runtimeId == "linux-x64") archSuffix = ".x86_64";
+                    }
+                    if (string.IsNullOrEmpty(archSuffix)) throw new NotSupportedException();
+
+                    var searchForTestFileDirectory = ArtifactsDirectory / packageType;
+                    Logger.Info($"Searching for files in {searchForTestFileDirectory}");
+                    var packageFile = searchForTestFileDirectory.GlobFiles($"*{archSuffix}.{packageType}")
+                        .Single();
+                    Logger.Info($"Testing Linux package file {packageFile}");
+
+                    var testScriptsBindMountPoint = RootDirectory / "linux-packages" / "test-scripts";
+
+                    DockerTasks.DockerPull(settings => settings.SetName(dockerImage));
+                    DockerTasks.DockerRun(settings => settings
+                        .EnableRm()
+                        .EnableTty()
+                        .SetImage(dockerImage)
+                        .SetEnv(
+                            $"VERSION={OctoVersionInfo.FullSemVer}",
+                            "INPUT_PATH=/input",
+                            "OUTPUT_PATH=/output",
+                            "SIGN_PRIVATE_KEY",
+                            "SIGN_PASSPHRASE",
+                            "REDHAT_SUBSCRIPTION_USERNAME",
+                            "REDHAT_SUBSCRIPTION_PASSWORD",
+                            $"BUILD_NUMBER={OctoVersionInfo.FullSemVer}")
+                        .SetVolume(
+                            $"{testScriptsBindMountPoint}:/test-scripts:ro",
+                            $"{ArtifactsDirectory}:/artifacts:ro")
+                        .SetArgs("bash", "/test-scripts/test-linux-package.sh", $"/artifacts/{packageType}/{packageFile}"));
+                });
+            }
+            
+            foreach (var testConfiguration in TestOnLinuxDistributions)
+            {
+                RunLinuxPackageTestsFor(testConfiguration);
+            }
+        });
+
+    // ReSharper disable once UnusedMember.Local
+    Target TestWindowsInstallerPermissions => _ => _
```

Comment:
I'm confused why this is in here... Why isn't it a test???
If that's out of scope, which i suspect it is, what if we pulled this out to a separate class?